src/PagerViewAdapter.tsx

```diff
@@ -72,8 +79,14 @@ export default function PagerViewAdapter<T extends Route>({
     }
 
     if (indexRef.current !== index) {
-      pagerRef.current?.setPage(index);
+      if (animationEnabled) {
+        pagerRef.current?.setPage(index);
+      } else {
+        pagerRef.current?.setPageWithoutAnimation(index);
+        position.setValue(index);
+      }
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Don't disable the lint rule. The lint rule is valid - the code in this function will break if `animationEnabled` isn't in the dependency array as it'll only respect what was set initially.

You can instead use `use-latest-callback` package for the `useCallback`, and include the dependency in the dep array for the `useEffect`.

Reply:
This comment should be removed